### PR TITLE
SD-1340 Add missing timestamps to join tables

### PR DESCRIPTION
**Pull request by tomurb**
Created with the following code:
```
created=ApplicationRecord.descendants.select do |x|
  !x.new.attributes.keys.include?('created_at')
rescue NotImplementedError
end.map(&:name)

updated=ApplicationRecord.descendants.select do |x|
  !x.new.attributes.keys.include?('updated_at')
  rescue NotImplementedError
end.map(&:name)

created.map! do |class_name|
  class_name.constantize.table_name
end

updated.map! do |class_name|
  class_name.constantize.table_name
end

created.uniq.each do |table_name|
  puts "add_column :#{table_name}, :created_at, :datetime"
end

updated.uniq.each do |table_name|
  puts "add_column :#{table_name}, :updated_at, :datetime"
end
```

### CODE REVIEW

File: core/db/migrate/20210702112334_add_missing_timestamp_columns.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class AddMissingTimestampColumns < ActiveRecord::Migration[6.1]
```
Comment (MatthewKennedy) [inline core/db/migrate/20210702112334_add_missing_timestamp_columns.rb:1]: @tomurb ,

We will need to use `[5.2]` here for the Rails 5.2 compatibility.
`class AddMissingTimestampColumns < ActiveRecord::Migration[5.2]`

Comment (tomurb) [reply core/db/migrate/20210702112334_add_missing_timestamp_columns.rb]: @MatthewKennedy fixed

**Comment by MatthewKennedy** [comment]
@tomurb  Looks good to me.

File: core/db/migrate/20210702112334_add_missing_timestamp_columns.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    add_column :spree_promotion_action_line_items, :updated_at, :datetime
+    add_column :spree_option_type_prototypes, :updated_at, :datetime
+    add_column :spree_order_promotions, :updated_at, :datetime
+    add_column :spree_product_promotion_rules, :updated_at, :datetime
+    add_column :spree_role_users, :updated_at, :datetime
+    add_column :spree_shipping_method_zones, :updated_at, :datetime
+    add_column :spree_option_value_variants, :updated_at, :datetime
+    add_column :spree_property_prototypes, :updated_at, :datetime
+    add_column :spree_prototype_taxons, :updated_at, :datetime  end
+end
```
Comment (damianlegawiec) [inline core/db/migrate/20210702112334_add_missing_timestamp_columns.rb:37]:
```suggestion
    add_column :spree_prototype_taxons, :updated_at, :datetime  
  end
end
```

File: core/db/migrate/20210702112334_add_missing_timestamp_columns.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class AddMissingTimestampColumns < ActiveRecord::Migration[5.2]
+  def change
+    # Add missing created_at columns
+    add_column :spree_promotion_actions, :created_at, :datetime
```
Comment (damianlegawiec) [inline core/db/migrate/20210702112334_add_missing_timestamp_columns.rb:4]: @tomurb also let's use a condition here, eg.

```suggestion
    add_column :spree_promotion_actions, :created_at, :datetime
    add_column :spree_promotion_actions, :created_at, :datetime unless column_exists?(:spree_promotion_actions, :created_at)
```

**Review by szymoniwacz — APPROVED** [review]

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**